### PR TITLE
Add Files.Pos + build/run scripts

### PR DESCRIPTION
**Pull request by DosWorld**
Hello!

I am find missed Files.Pos procedure (like 'tell' in C, 'FilePos' in Pascal).
Also, i am change type of offset in procedure Seek to LONGINT (to be able process huge files)

About second commit (i don't have a plan merge it to your repository, but may be you have a look).
It was non-clean for me about classpath and distribution result software.
So, i make changes: all (compiler and runtime) move to ob.jar and create shell scripts
oc and orun (compile, run) for linux.
I also make changes for windows (oc.bat orun.bat), but i have no windows and can't test it.

PS: I am want (at end) receive Oberon compiler for ms-dos, so i want extend oberonc to be able compile Oberon compiler from Project Oberon site. I see some non-compability into API (like Files etc). I can contibute this changes to your repository. Did you  want receive it?

(sorry, english is not my native language)

### CODE REVIEW

File: src/java/Files.java [hunk 1]
```diff
@@ -443,30 +443,34 @@ public static int ReadStr(Files_FileDesc file, char[] str) {
...
   }
 
-  public static int Seek(Files_FileDesc file, int pos) {
-    int r;
-    r = OK;
+  public static int Seek(Files_FileDesc file, long pos) {
+    file.err = OK;
     try {
       file.f.seek(pos);
-      if(file.err == EOF) {
```
Comment (lboasso) [inline src/java/Files.java:451]: Why do you want to remove these lines?

File: src/java/Files.java [hunk 2]
```diff
@@ -443,30 +443,34 @@ public static int ReadStr(Files_FileDesc file, char[] str) {
...
-        file.err = OK;
-      }
     } catch(IOException e) {
-      r = IOERROR;
+      file.err = IOERROR;
```
Comment (lboasso) [inline src/java/Files.java:451]: Setting `file.err = IOERROR;` means that this file will be considered invalid. We should not invalidate the file if the caller writes `Seek(f, -1)` for example.

File: src/java/Files.java [hunk 3]
```diff
@@ -443,30 +443,34 @@ public static int ReadStr(Files_FileDesc file, char[] str) {
...
     }
-    return r;
+    return file.err;
+  }
+
+  public static long Pos(Files_FileDesc file) {
```
Comment (lboasso) [inline src/java/Files.java:456]: `oberonc` support only integers not longs, so we cannot call this from Oberon code.

**Review by lboasso — COMMENTED** [review]
Hello, thanks for your interest in oberonc. I left some comments.
You might be interested in this project http://oberon.wikidot.com/project-oberon-java

**Comment by DosWorld** [comment]
Hello!

Problems has been fixed.

PS: I'm interested in "Oberon without the Project Oberon". Thank you, your link can help in mapping between Oberon and Java

File: src/java/Files.java [hunk 4]
```diff
@@ -457,12 +457,24 @@ public static int Seek(Files_FileDesc file, int pos) {
...
+    return -1;
+  }
+
+  public static int Length(Files_FileDesc file) {
     int len;
     len = -1;
     if(file.err == OK) {
       try {
-        len = (int) file.f.length();
+        long p = file.f.length();
```
Comment (lboasso) [inline src/java/Files.java:476]: The original code `len = (int) file.f.length();` is correct and does the same.

File: src/java/Files.java [hunk 5]
```diff
@@ -457,12 +457,24 @@ public static int Seek(Files_FileDesc file, int pos) {
...
+      try {
+        long p = file.f.getFilePointer();
+        return (int)(p > Integer.MAX_VALUE ? Integer.MAX_VALUE : p);
+      } catch(IOException e) {
+      }
+    }
+    return -1;
+  }
+
+  public static int Length(Files_FileDesc file) {
```
Comment (lboasso) [inline src/java/Files.java:471]: I prefer not to rename this function.

**Review by lboasso — COMMENTED** [review]
Why do you need this functionality? In general `Files.Mod` is for the compiler use only, I prefer to keep it as small as possible with only the functionality needed by the compiler. You are free to create your own file library to use in your project.

**Comment by DosWorld** [comment]
Yep, looks like try to push it to oberonc - is not nice idea.

**Pull request closed without merging**